homeassistant/components/ecovacs/entity.py

```diff
@@ -120,3 +121,35 @@ class EcovacsCapabilityEntityDescription(
     """Ecovacs entity description."""
 
     capability_fn: Callable[[Capabilities], CapabilityEntity | None]
+
+
+class EcovacsLegacyEntity(Entity):
+    """Ecovacs legacy bot entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, device: VacBot) -> None:
+        """Initialize the legacy Ecovacs entity."""
+        self.device = device
+        vacuum = device.vacuum
+
+        self.error: str | None = None
+        self._attr_unique_id = vacuum["did"]
+
+        if (name := vacuum.get("deviceName")) is None:
+            name = vacuum.get("name", vacuum["did"])
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, vacuum["did"])},
+            model=f"{vacuum['product_category']} {vacuum['model']}",
```

Comment:
Also, what are either value? Is any of them fit for the new model_id?

Reply:
these are the values:
```json
        "deviceName": "DEEBOT Slim2 Series",
        "product_category": "DEEBOT",
        "model": "Slim2",
```

The device is a "DEEBOT Slim2" not sure if this meets the model_id criteria, but i think "DEEBOT Slim2" sounds not technical enough  :thinking:

Reply:
when i rename the bot in the app, than `deviceName` will be changed, so `product_category` plus `model` will represent the full model name

Reply:
mehhh ... `deviceName` keeps as is when changing the name ... will check again

Reply:
ok ... after some testes with the app and the returned data:

`name` ist always the same as `did` (_which is the serial number printed on the type label on the device_)
`deviceName` is the "human readable" model description
`nick` is the name of the bot in case you set one in the app